tools/kagami/src/swarm.rs

```diff
@@ -256,8 +261,14 @@ impl PrepareConfigurationStrategy {
 
                 let spinner = ui.spinner_validator();
 
-                let validator = super::validator::construct_validator()
-                    .wrap_err("Failed to construct the validator")?;
+                let validator = match source {
+                    ResolvedImageSource::Build { ref path } => {
+                        super::validator::construct_validator(Some(path))
+                            .wrap_err("Failed to construct the validator")?
+                    }
+                    _ => super::validator::construct_validator(None)
+                        .wrap_err("Failed to construct the validator")?,
+                };
```

Comment:
Hmmph, it doesn't work this way.

Whether `kagami swarm` should build the validator or not depends not on the source of image (`build`, `build-from-github` or `image`), but on `--no-default-configuration` option. The validator is a part of the configuration in `<dir>/config` directory, which is volumed in `docker-compose.yml` and does not depend on image source.

So, generating the validator or not directly depends on `self`: `Self::GenerateDefault` means to generate the configuration, including the validator, and `Self::GenerateOnlyDirectory` means... what it says.

Reply:
Maybe it should be put into a docstring, though

Reply:
We're doing it inside of a `Self::GenerateDefault` match arm if you double check above the diff. The distinction here is whether we need to clone `iroha` behind the scenes (the arm with `None` passed to the `construct_validator` func) or if we have the shallow clone already.

Reply:
Indeed, I am blind, my apologies.